internal/database/mysql.go

```diff
@@ -103,7 +103,10 @@ func (db *MySQLDatabase) AuthenticateLogin(username string, password string) (id
 	).Scan(&id, &expectedPassword)
 	if errors.Is(err, sql.ErrNoRows) {
 		return 0, errors.New("invalid username")
-	} else if expectedPassword != password {
+	} 
+	err = bcrypt.CompareHashAndPassword([]byte(expectedPassword), []byte(password))
+	log.Printf("Password: %s, Expected Password: %s\n", password, expectedPassword)
```

Comment:
use [log.Println("Password:", password, "Expected Password:", expectedPassword)] since we want to minimize use of string formatting for necessary sql string generation.

However, I don't think we should be logging hashed passwords. It would be much better imo to:
if err != nil {
    log.Println("Failed to verify user password -", err)
}

And simply print nothing if it works.

Reply:
Understood, I have removed the print as it is unecessary.